File: templates/go-service/service/components/setting.go
package components

import (
	"app/context"
	"app/transport"

	"github.com/gin-gonic/gin"
)

// Setting is data to be injected into SetUp
type Setting struct {
	Ctx        *context.Context
	Router     *gin.Engine
	Publishers struct {
		Main transport.Publisher
	}
	Subscribers struct {
		Main transport.Subscriber
	}
	RPCServers struct {
		Main      transport.RPCServer
		Secondary transport.RPCServer
	}
	RPCClients struct {
		MainLoopBack      transport.RPCClient
		SecondaryLoopBack transport.RPCClient
	}
}

File: templates/nodejs-service/service/src/components/setting.ts
import express from "express";
import { Express } from "express";
import { Context } from "../context";
import { Publisher, Subscriber, RPCServer, RPCClient, RmqPublisher, RmqSubscriber, RmqRPCServer, RmqRPCClient } from "../transport";

export class Setting {
    static globalSetting: Setting;

    ctx: Context;
    router: Express;
    publishers: Publishers;
    subscribers: Subscribers;
    rpcServers: RPCServers;
    rpcClients: RPCClients;

    constructor(ctx: Context, router: Express, publishers: Publishers, subscribers: Subscribers, rpcServers: RPCServers, rpcClients: RPCClients) {
        if (Setting.globalSetting) {
            throw new Error("setting initialized, user Setting.getInstance() instead");
        }
        this.ctx = ctx;
        this.router = router;
        this.publishers = publishers;
        this.subscribers = subscribers;
        this.rpcServers = rpcServers;
        this.rpcClients = rpcClients;
        Setting.globalSetting = this;
    }

    getInstance() {
        return Setting.globalSetting;
    }
}

export interface Publishers {
    main: Publisher;
}

export interface Subscribers {
    main: Subscriber;
}

export interface RPCServers {
    main: RPCServer;
    secondary: RPCServer;
}

export interface RPCClients {
    mainLoopBack: RPCClient;
    secondaryLoopBack: RPCClient;
}